src/uct/tcp/tcp_cm.c

```diff
@@ -74,6 +74,7 @@ void uct_tcp_cm_change_conn_state(uct_tcp_ep_t *ep,
             /* Since ep::peer_addr is 0'ed, we have to print w/o peer's address */
             full_log = 0;
         }
+        uct_tcp_ep_mod_events(ep, 0, UCS_EVENT_SET_EVWRITE);
```

Comment:
@dmitrygx is this correct?

Reply:
if I'm not mistaken, all requested epoll events have to be removed before moving EP to CLOSED connection state.
maybe add `ucs_assert_always(ep->conn_state == 0)` here and check why this is not zero.